cluster-autoscaler/processors/provreq/processor.go

```diff
@@ -108,5 +121,40 @@ func (p *provReqProcessor) Process(provReqs []*provreqwrapper.ProvisioningReques
 	}
 }
 
-// Cleanup cleans up internal state.
+// CleanUp cleans up internal state
 func (p *provReqProcessor) CleanUp() {}
+
+// BookCapacity schedule fake pods for ProvisioningRequest that should have reserved capacity
+// in the cluster.
+func (p *provReqProcessor) BookCapacity(ctx *context.AutoscalingContext) error {
```

Comment:
I think this should be implemented as PodListProcessor.Process() rather than introducing a new call to StaticAutoscaler:
 * It's literally doing what PLP is meant to do - changing the list of pods to be processed by CA (by injecting new ones). 
 * Scheduling pods on existing nodes is also generally done in PLP - [FilterOutSchedulable](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/core/podlistprocessor/filter_out_schedulable.go) is the main place we do it.
 * Injecting pods in PLP in a similar way is a pretty well established pattern in CA forks - I know you have access to GKE fork, you can see how CapacityRequests do pretty much the same thing in a PLP.

Reply:
PodListProcessor is processing the list of unschedulable pods and applying changes to it. Since booking capacity does nothing to unschedulable pods list and just modifying cluster snapshot. 
Sure, I can implement booking capacity as a part of PodListProcessor and just do nothing with unschedulable list, is it what you asking? However I don't see the advantage of this approach.